Makefile

```diff
@@ -55,15 +55,25 @@ ifeq ($(go_bindata),)
 go_bindata := cd hack/ci-deps && go install github.com/kevinburke/go-bindata/... && cd ../.. && "${GOPATH}/bin/go-bindata"
 endif
 
-GOLANG_IMAGE = golang:1.16-alpine
-GO ?= GOCACHE=/tmp/.cache docker run --rm -v "$(CURDIR)":/go/src/github.com/k0sproject/k0s \
+ifeq ($(TARGET_OS),windows)
+build: k0s.exe
+else
+build: k0s
+endif
+
+golang_image := $(shell docker inspect --type=image k0sbuild.docker-image.k0s)
+ifneq ($(golang_image),)
+golang_image := docker build --rm -t k0sbuild.docker-image.k0s -f build/Dockerfile .
+endif
+
+GO ?= $(golang_image) && GOCACHE=/tmp/.cache docker run --rm -v "$(CURDIR)":/go/src/github.com/k0sproject/k0s \
 	-w /go/src/github.com/k0sproject/k0s \
 	-e GOOS \
 	-e CGO_ENABLED \
 	-e GOARCH \
 	-e GOCACHE \
 	--user $$(id -u) \
-	$(GOLANG_IMAGE) go
+	k0sbuild.docker-image.k0s go
```

Comment:
Can you put back the `GOLANG_IMAGE` variable?  This is used heavily by other out-of-band build processes.

As for the `BUILD_*` changes, those are okay.

Reply:
see my comment below about the custom Golang image that is now used.

Reply:
> Can you put back the `GOLANG_IMAGE` variable? This is used heavily by other out-of-band build processes.

can you point me to where this is being used?

Reply:
I see your notes about the custom image.

What about building `k0sbuild.docker-image.k0s` as a target that is depended on by `build`.  That would allow for easier extending.  Something like:

```
GOLANG_IMAGE = k0sbuild.docker-image.k0s
GO ?= GOCACHE=/tmp/.cache docker run --rm -v "$(CURDIR)":/go/src/github.com/k0sproject/k0s \
	-w /go/src/github.com/k0sproject/k0s \
	-e GOOS \
	-e CGO_ENABLED \
	-e GOARCH \
	-e GOCACHE \
	--user $$(id -u) \
	$(GOLANG_IMAGE) go

.k0sbuild.docker-image.k0s:
	docker build --rm -t k0sbuild.docker-image.k0s -f build/Dockerfile .
	touch $@

.PHONY: build
ifeq ($(TARGET_OS),windows)
build: .k0sbuild.docker-image.k0s k0s.exe
else
build: .k0sbuild.docker-image.k0s k0s
endif
```